scripts/analysis/get-all-components.ts

```diff
@@ -0,0 +1,138 @@
+import { join } from 'path';
+import { readdir, stat, readdirSync, readFileSync } from 'fs';
+import { promisify } from 'util';
+import { packageNames } from '../package-names';
+import { paths } from '../paths';
+import { parse } from '@typescript-eslint/typescript-estree';
+
+const readdirP = promisify(readdir);
+const statP = promisify(stat);
+
+const coreComponents = join(__dirname, '../../src/core/components');
+const editorialComponents = join(
+	__dirname,
+	'../../src/editorial/web/components',
+);
+
+const isDirectory = (path: string) =>
+	statP(path).then((stats) => stats.isDirectory());
+
+const getComponentPackageNamesAndPaths = () =>
+	Promise.all([
+		readdirP(coreComponents)
+			.then((componentDirs) =>
+				Promise.all(
+					componentDirs.map((componentDirName) =>
+						isDirectory(
+							`${coreComponents}/${componentDirName}`,
+						).then((isDir) => {
+							if (!isDir) return;
+
+							return {
+								[`@guardian/src-${componentDirName}`]: `${coreComponents}/${componentDirName}`,
+							};
+						}),
+					),
+				),
+			)
```

Comment:
I know it's a short-lived thing, but we could lift this into `package-names.ts` or make use of the existing `getComponentPackageNames` function like so:

```suggestion
		getComponentPackageNames().then(packageNames => {
		  return packageNames.reduce((acc, pn) => {
		    return {
		      ...acc,
		      [pn]: `${coreComponents}/${componentDirName}`
		    }
		  })
		})
```

Reply:
I've made some changes in the latest commit to shift stuff out into both `paths.ts` and `package-names.ts`. How does it look now?